tests/ffi/qm_nested/test.sh

```diff
@@ -0,0 +1,50 @@
+#!/bin/bash
+
+# shellcheck disable=SC1091
+. ../common/prepare.sh
+
+# Declare global variables and initialize to 0
+NESTED_NAME=0
+NESTED_STATUS=0
+
+# Create nested container fedora inside the QM.
+# Function to create and start nested container fedora, while executing podman commands from inside the QM container.
+# Extract nested container's valuable information and store it inside the global variables for later use.
+# Globals:
+#  NESTED_NAME: The nested container name in the QM container.
+#  NESTED_STATUS: The nested container status in the QM container.
+create_nested() {
+        podman exec -it qm bash -c "podman run -d -it --name fedora fedora /bin/bash"
+        echo "Container started inside QM."
+        NESTED_NAME=$(podman exec -it qm bash -c "podman inspect fedora --format '{{.Name}}'")
+        echo "Container name is: $NESTED_NAME"
+        NESTED_STATUS=$(podman exec -it qm bash -c "podman inspect --format='{{.State.Status}}' fedora")
+        echo "Container $NESTED_NAME status is: $NESTED_STATUS"
+}
+
+# Function to check and verify that nested container fedora was created and is running inside the QM container.
+# Globals:
+#  NESTED_NAME: The nested container name in the QM container.
+#  NESTED_STATUS: The nested container status in the QM container.
+compare_values() {
+        local expected_nested_name="fedora"
+        local expected_nested_status="running"
+
+        if [ "$NESTED_NAME" == 0 ] || [ "$NESTED_STATUS" == 0 ]; then
+                echo "FAIL: Failed to run the container inside QM."
+                exit 1
+        elif [[ $NESTED_NAME != *"$expected_nested_name"* ]];then
```

Comment:
Please use shellcheck tool against your script, I am sure it will complain about not using "" around this variable.

Example:

"$NESTED_NAME" or even better "${NESTED_NAME}".  The ${var} is used when you need to reference the variable in a context where the variable name might be followed by characters that could confuse the shell or where additional operations are needed on the variable. In other words, ${var} is always preferable to my eyes at least as it's safer.

Reply:
I indeed used the shellcheck, no complains were received.

ShellCheck Output
$ shellcheck myscript
No issues detected!

$